contracts/protocol/libs/FundsLib.sol

```diff
@@ -217,14 +229,56 @@ library FundsLib {
         }
 
         if (payoff.protocol > 0) {
-            increaseAvailableFunds(0, exchangeToken, payoff.protocol);
+            increaseAvailableFunds(PROTOCOL_ENTITY_ID, exchangeToken, payoff.protocol);
             emit IBosonFundsLibEvents.ProtocolFeeCollected(_exchangeId, exchangeToken, payoff.protocol, sender);
         }
         if (payoff.agent > 0) {
             // Get the agent for offer
             uint256 agentId = ProtocolLib.protocolLookups().agentIdByOffer[exchange.offerId];
             increaseAvailableFundsAndEmitEvent(_exchangeId, agentId, exchangeToken, payoff.agent, sender);
         }
+        BosonTypes.DisputeResolutionTerms storage drTerms = pe.disputeResolutionTerms[offer.id];
+        if (payoff.disputeResolver > 0) {
+            increaseAvailableFundsAndEmitEvent(
+                _exchangeId,
+                drTerms.disputeResolverId,
+                exchangeToken,
+                payoff.disputeResolver,
+                sender
+            );
+        }
+
+        // Return unused DR fee to mutualizer or seller's pool
+        if (drTerms.feeAmount != 0) {
+            uint256 returnAmount = drTerms.feeAmount - payoff.disputeResolver;
+
+            if (drTerms.mutualizerAddress == address(0)) {
+                if (returnAmount > 0) {
+                    increaseAvailableFundsAndEmitEvent(
+                        _exchangeId,
+                        offer.sellerId,
+                        exchangeToken,
+                        returnAmount,
+                        sender
+                    );
+                }
+            } else {
+                if (returnAmount > 0) {
+                    decreaseAvailableFunds(PROTOCOL_ENTITY_ID, exchangeToken, returnAmount);
```

Comment:
In connection with my comment in exchangeHandler - if we are not increasing funds there, we also should not decrease theme here.